Rationale.md

```diff
@@ -0,0 +1,201 @@
+  [garbage collection](GC.md).
+
+
+## Load/Store Addressing
+
+Load/store instructions include an immediate offset used for
+[addressing](AstSemantics.md#Addressing). This is intended to simplify folding
+of offsets into complex address modes in hardware, and to simplify bounds
+checking optimizations. It offloads some of the optimization work to the
+compiler that targets WebAssembly, executing on the developer's machine, instead
+of performing that work in the WebAssembly compiler on the user's machine.
+
+
+## Alignment Hints
+
+Load/store instructions contain
+[alignment hints](AstSemantics.md#Alignment). This makes it easier to generate
+efficient code on certain hardware architectures.
+
+Either tooling or an explicit opt-in "debug mode" in the spec could allow
+execution of a module in a mode that threw exceptions on misaligned access.
+This mode would incur some runtime cost for branching on most platforms which is
+why it isn't the specified default.
+
+
+## Out of Bounds
+
+The ideal semantics is for
+[out-of-bounds accesses](AstSemantics.md#Out-of-Bounds) to trap, but the
+implications are not yet fully clear.
+
+There are several possible variations on this design being discussed and
+experimented with. More measurement is required to understand the associated
+tradeoffs.
+
+  * After an out-of-bounds access, the instance can no longer execute code and
+    any outstanding JavaScript [ArrayBuffer][] aliasing the linear memory are
+    detached.
+    * This would primarily allow hoisting bounds checks above effectful
+      operations.
+    * This can be viewed as a mild security measure under the assumption that
+      while the sandbox is still ensuring safety, the instance's internal state
+      is incoherent and further execution could lead to Bad Things (e.g., XSS
+      attacks).
+  * To allow for potentially more-efficient memory sandboxing, the semantics
+    could allow for a nondeterministic choice between one of the following when
+    an out-of-bounds access occurred.
+    * The ideal trap semantics.
+    * Loads return an unspecified value.
+    * Stores are either ignored or store to an unspecified location in the
+      linear memory.
+    * Either tooling or an explicit opt-in "debug mode" in the spec should allow
+      execution of a module in a mode that threw exceptions on out-of-bounds
+      access.
+
+  [ArrayBuffer]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer
+
+
+## Resizing
+
+Implementations provide a `page_size` operation which allows them to efficiently
+map the underlying OS's capabilities to the WebAssembly application, as well as
+to communicate their own implementation details in a useful manner to the
+developer.
+
+
+## Control Flow
+
+See [#299](https://github.com/WebAssembly/design/pull/299).
+
+
+## Locals
+
+Address-taken variables in the call stack are expected to be in linear memory,
```

Comment:
I know this is a copy-over, but this might be a good opportunity to revise this section to make clear that address-taken is a concept only in certain languages, e.g. C/C++.